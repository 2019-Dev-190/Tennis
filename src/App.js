import React, { Component } from 'react';
import './assets/css/gameTennis.css';
import FirstPlayer from './components/firstPlayer/firstPlayer';
import SecondPlayer from './components/secondPlayer/secondPlayer';
import ScoreBoard from './components/scoreBoard/scoreBoard';


export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstPlayer: 0,
      secondPlayer: 0,
      gameBall: false,
      firstPlayerAdvantage: false,
      secondPlayerAdvantage: false,
      won: 0,
      deuce: false,
      points: [0, 15, 30, 40, 'won']
    };
  }


  updateScorePlayer = (player) => {
    const { deuce, gameBall, firstPlayerAdvantage, secondPlayerAdvantage, firstPlayer, secondPlayer, points } = this.state;

    if(player === 1 && !this.state.won)
    {
      const firstPlayer = this.state.firstPlayer + 1;

      if(gameBall && !secondPlayerAdvantage)
      {
        this.setState({won: 1});
      }
      if (deuce) {
        if (firstPlayerAdvantage) {
          this.setState({ gameBall: true });
        } else {
          this.setState({
            secondPlayerAdvantage: false,
            firstPlayerAdvantage: true,
          });
        }
      } else {

        if (points[firstPlayer] === 'won') {
          this.setState({ won: 1 });
        } else if (points[firstPlayer] === 40 && points[secondPlayer] === 40) {
          this.setState({
            deuce: true,
            firstPlayerAdvantage: true,
            firstPlayer
          });
        } else {
          this.setState({ firstPlayer });
        }
      }
  }
  }



  render() {
    const { firstPlayer, secondPlayer, won, points } = this.state;

    return (
      <div className="App">
        <div className="tennisBg">
          <h2 className="THeading">Let's Play Tennis</h2>
          <div>
            <ScoreBoard firstPlayer={points[firstPlayer]} secondPlayer={points[secondPlayer]} />
          </div>
          <div className="playerBox">
            <FirstPlayer updateScore={this.updateScorePlayer} />
            <SecondPlayer updateScore={this.updateScorePlayer} />
          </div>
          {
            won &&
            <h1>Player {won} Won!</h1>
          }
        </div>
      </div>
    );
  }
}
