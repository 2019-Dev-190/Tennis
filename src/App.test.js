import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FirstPlayer from './components/firstPlayer/firstPlayer';

configure({ adapter: new Adapter() });


describe('App', ()=>{
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('handle first player first ball call "updateScore" prop on button click', () => {
  const updateScore = jest.fn();
  const button = shallow((<FirstPlayer updateScore={updateScore} />));
  button.find('.player1').props().onClick();
  expect(updateScore).toHaveBeenCalled();
  expect(updateScore).toHaveBeenCalledTimes(1);
});

});
