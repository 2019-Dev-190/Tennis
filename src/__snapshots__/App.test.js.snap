// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should update player score 1`] = `
<div
  className="App"
>
  <div
    className="tennisBg"
  >
    <h2
      className="THeading"
    >
      Let's Play Tennis
    </h2>
    <div>
      <div
        className="scoreBoard"
      >
        <h3>
          scoreBoard
        </h3>
        <div
          className="playerStyles"
        >
          Player1: 
          <div
            className="scoreStyles"
          >
            0
          </div>
        </div>
        <div
          className="playerStyles"
        >
          Player2: 
          <div
            className="scoreStyles"
          >
            0
          </div>
        </div>
      </div>
    </div>
    <div
      className="playerBox"
    >
      <div>
        <div
          className="playerHeading"
        >
          Player1
        </div>
        <div
          className="player1"
          onClick={[Function]}
        />
      </div>
      <div>
        <div
          className="playerHeading"
        >
          Player2
        </div>
        <div
          className="player2"
        />
      </div>
    </div>
    0
  </div>
</div>
`;
